File: spring-microservices/identity-service/src/main/java/com/devteria/identity/dto/request/StudentRequest.java
package com.devteria.identity.dto.request;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StudentRequest {
    @NotBlank(message = "Username is required")
    private String username;

    private String password;

    @Email(message = "Email should be valid")
    @NotBlank(message = "Email is required")
    private String email;

    @NotBlank(message = "First name is required")
    private String firstName;

    @NotBlank(message = "Last name is required")
    private String lastName;

    private String dob;

    @NotBlank(message = "Student ID is required")
    private String studentId;

    private String departmentId;

    private String educationalUnitId;

    private String description;

    private String socialUrl;

    private String className;
}

File: react-type-vite/src/services/api/request/studentRequest.ts
export interface StudentRequest {
  username: string;
  password: string;
  email: string;
  firstName: string;
  lastName: string;
  dob?: string;
  studentId: string;
  departmentId?: string;
  educationalUnitId: number;
  description?: string;
  socialUrl?: string;
  className?: string;
}